src/backends/interflop-mca-int/interflop_mca_int.c

```diff
@@ -165,30 +166,23 @@ static __thread rng_state_t rng_state;
 /* -126-24+-126-24 = -300 > DOUBLE_EXP_MIN (-1022) */
 static inline void _noise_binary64(double *x, const int exp,
                                    rng_state_t *rng_state) {
-  int64_t noise;
-  uint32_t shift;
-
   // Convert preserving-bytes double to int64_t
-  int64_t x_s64 = *(int64_t *)(x);
+  binary64 *b64 = (binary64 *)x;
 
   // amount by which to shift the noise term sign (1) + exp (11) + noise
   // exponent
-  shift = 1 + DOUBLE_EXP_SIZE - exp;
+  const uint32_t shift = 1 + DOUBLE_EXP_SIZE - exp;
 
   // noise is a signed integer so the noise is centered around 0
   // only 32 bits of noise are used, they are left aligned in a signed 64 bit
```

Comment:
What is the advantage of using 64 bits of noise (get_rand_uint64) here over 32 ? 
I'm fine with both but would love the explanation. In any case, if you keep 64 random bits could you please update comment in line 177 ?

Reply:
Since the RNG only generates 64 bits at once, it's the same cost to ask for 64 or 32 bits. I added the function to generate 32 random bits for convenience only.
I'll fix the comment. Thanks!